File: backend/app/schemas/activity.py
from pydantic import BaseModel

from typing import Optional

from datetime import datetime

from enum import Enum

class ActivityTypeEnum(str, Enum):

    call = "call"

    email = "email"

    note = "note"

    meeting = "meeting"

class ActivityCreate(BaseModel):

    type: ActivityTypeEnum

    note: Optional[str] = None

    lead_id: Optional[int] = None

    contact_id: Optional[int] = None

    deal_id: Optional[int] = None

class ActivityResponse(BaseModel):

    id: int

    type: ActivityTypeEnum

    note: Optional[str]

    created_by_id: Optional[int]

    lead_id: Optional[int]

    contact_id: Optional[int]

    deal_id: Optional[int]

    created_at: datetime

    class Config:

        from_attributes = True

File: frontend/types/activity.ts
export interface Activity {
  id: number;
  type: 'call' | 'email' | 'note' | 'meeting';
  note: string | null;
  created_by_id: number | null;
  lead_id: number | null;
  contact_id: number | null;
  deal_id: number | null;
  created_at: string;
}

export const ACTIVITY_ICONS: Record<string, string> = {
  call: '📞',
  email: '✉️',
  note: '📝',
  meeting: '🤝',
};

export const ACTIVITY_COLORS: Record<string, string> = {
  call: 'bg-blue-100 text-blue-700',
  email: 'bg-purple-100 text-purple-700',
  note: 'bg-gray-100 text-gray-700',
  meeting: 'bg-green-100 text-green-700',
};